yt/units/yt_array.py

```diff
@@ -1438,7 +1438,10 @@ def __setstate__(self, state):
         metadata extracted in __reduce__ and then serialized by pickle.
         """
         super(YTArray, self).__setstate__(state[1:])
-        unit, lut = state[0]
+        try:
+            unit, lut = state[0]
+        except TypeError:
```

Comment:
Maybe you should add a comment here to state that this make the very old pickle files work. Otherwise the code looks a bit cryptic.